int64[] tag_id
float64[] distance